tests/pytest.ini

```diff
@@ -5,6 +5,7 @@ addopts =
     --durations=10
     --showlocals
     -m 'not nonci and not no_block_pr'
+    --json-report --json-report-file=../test_results/test-report.json
```

Comment:
nit: Could you add a bit more context on the commit message, on what is this option, what effect does it have etc.?

Reply:
Fixed, but let me know if you have questions. It may be easier to just show you an example report.